libraries/engage/tracking/reducers/cookieSettings.js

```diff
@@ -16,7 +19,9 @@ const cookieSettingsReducer = (state = defaultState, action) => {
     case UPDATE_COOKIE_CONSENT: {
       return {
         ...state,
-        comfortCookiesAccepted: action.comfortCookiesAccepted,
+        comfortCookiesAccepted: showComfortCookiesToggle === true
```

Comment:
doing this here has some disadvantages. the streams emit based on the action payload, but this would still be `false` in case of disabled comfort cookies. so they would reflect the wrong state. additionally the system would initialize wrong for people who already have a "wrong" state saved in Redux since they already handled the cookie consent.

i think instead of the reducer, you need to add this logic to the `updateCookieConsent` action creator and the `getAreComfortCookiesAccepted` selector.

the action creator is used for all consent state updates triggered by user interaction. it triggers the streams that can be used by e.g. extensions to register for those updates.

the selector is used at app start to retrieve the values for the `handleCookieConsent` action. additionally it can be used by extensions to determine the current comfort cookies consent.

since this bug is not related to the ticket number from your branch name, please create a new one with a different ticket number.

Reply:
fixed, please check again